src/twisted/internet/defer.py

```diff
@@ -927,6 +928,48 @@ def main(reactor):
     return coro
 
 
+def coroFnToDeferredFn(corofn):
+    """
+    L{coroFnToDeferredFn} helps you define coroutinefunctions where previously
+    there were functions that returned Deferreds
+
+    Coroutine functions return a coroutine object, similar to how generators
+    work. This function turns that coroutinefunction into a Deferred function,
+    meaning that it can be used in legacy Twisted code. For example::
+
+        import treq
```

Comment:
can we have an example that is using only twisted and don't rely on an external library

```suggestion
        from twisted.web.client import getPage
```

Reply:
this is copied from ensureDeferred:

https://github.com/twisted/twisted/blob/a71a78145a59a4dea61a84d37c22bb738c9a4224/src/twisted/internet/defer.py#L898-L903